EIPS/eip-ondemand-data.md

```diff
@@ -0,0 +1,152 @@
+---
+title: On-Demand Offchain Data Retrieval
+description: A standard to construct multicalls with prepended verifiable off-chain data
+author: Noah Litvin (@noahlitvin), db (@dbeal-eth)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-26
+---
+
+## Abstract
+
+Contracts may require off-chain data that is not already available on chain during execution. A smart contract function could implement the standard proposed here by reverting with `error OracleDataRequired(bytes32 oracleId, bytes oracleQuery, address oracleContract)`. Clients supporting this standard would recognize this error message during a simulation of the request, query the specified decentralized oracle network for signed data, and instead stage a transaction with a multicall that prepends the verification of the required off-chain data. The data would be written on-chain during verification to a smart contract for the subsequent call to read, avoiding the error.
```

Comment:
Afaik multicalls are not a standard in Ethereum. So it would be good if you refer the multicall contract and explain it a bit.